apps/hash-frontend/src/shared/layout/layout-with-sidebar/account-entity-type-list/search-input.tsx

```diff
@@ -0,0 +1,98 @@
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import {
+  FontAwesomeIcon,
+  IconButton,
+  TextField,
+} from "@hashintel/design-system";
+import { Fade, outlinedInputClasses, Tooltip } from "@mui/material";
+import { FunctionComponent } from "react";
+
+import { MagnifyingGlassLightIcon } from "../../../icons/magnifying-glass-light";
+
+type SearchInputProps = {
+  searchVisible: boolean;
+  showSearchInput: () => void;
+  hideSearchInput: () => void;
+  onChangeText: (text: string) => void;
+};
+
+export const SearchInput: FunctionComponent<SearchInputProps> = ({
```

Comment:
UX question to consider (not for this PR) for @nonparibus – now that we have the 'view all types', and this search input appears by clicking a button next to 'view all types', it's unexpected that searching in the input just searches the list of account types above. We should think about what to do with this search bar, whether that's to launch into the table with a filter, or do a dynamic query to the db and show the results above, or something else

In this row I can click 'view all types', or 'search types'
![Screenshot 2023-08-15 at 17 40 10](https://github.com/hashintel/hash/assets/37743469/0994eb5b-7525-499e-8363-de14dc370a6e)

'search types' doesn't search all types, but just the list of types in my account above
![Screenshot 2023-08-15 at 17 40 19](https://github.com/hashintel/hash/assets/37743469/d9291027-00a8-4c9e-9e4c-d901d232ca77)

Reply:
Will respond separately. Tracking this UX Q as H-418.